generate/main.go

```diff
@@ -0,0 +1,205 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"regexp"
+	"strings"
+)
+
+// ServerVersion : the version from the upstream repository
+var ServerVersion = getDefaultVersion()
+
+// GrafeasRepository : the repository of the Grafeas server to use for downloading Swagger files
+var GrafeasRepository = "https://github.com/grafeas/grafeas"
+
+// Reference : the commit, branch, or tag to use for downloading Swagger files from the defined Grafeas repo
+var Reference = "master"
+
+// APIVersion : the version of the Grafeas API per the Google Cloud scheme - https://github.com/grafeas/grafeas/blob/master/docs/versioning.md#api
+var APIVersion = "v1beta1"
+
+// SwaggerCodegenVersion : the version of the Swagger CodeGen CLI to download
+var SwaggerCodegenVersion = "2.4.5"
+
+// MergedClient : whether to keep the generated Swagger clients separate or to merge the paths
+// TODO: implement function that merges the paths of each Swagger spec into one file before running codegen
```

Comment:
I'm not sure I follow this: what would the merge look like of each Swagger spec? What's the end result?

Reply:
**TL;DR** I took the liberty of pulling in the project API spec, and thought it might be nice to merge it with the grafeas spec at some point.

I hadn't taken the time to implement it yet, but the idea was (I'm sure you can speak to the validity of this more than I can) was to have a more uniform "client" experience. Right now, the code generation was _only_ using [grafeas.swagger.json](https://raw.githubusercontent.com/grafeas/grafeas/master/proto/v1beta1/swagger/grafeas.swagger.json) from the upstream [grafeas](https://github.com/grafeas/grafeas) repository. Because of this, it is missing some of the project API interactions (e.g. create, delete, list, get). I think that this would be nice to have as part of the overarching client. So, at a minimum I have pulled in the `projects.swagger.json` from the upstream repo as part of the `go generate`
Define all swagger spec files
https://github.com/grafeas/client-go/pull/15/files#r379532306
Iterate over all swagger spec files
https://github.com/grafeas/client-go/pull/15/files#r379532718

With that in place the interaction looks something like
```go
package main

import (
	"context"
	"fmt"
	"log"

	grafeasAPI "github.com/psprings/client-go/0.1.4"
)

func main() {
	projectID := "projects/myproject"
	grafeasCfg := &grafeasAPI.Configuration{
		BasePath:      "http://localhost:8080",
		DefaultHeader: make(map[string]string),
		UserAgent:     "Swagger-Codegen/0.1.0/go",
	}
	grafeasClient := grafeasAPI.NewAPIClient(grafeasCfg)

	notesResp, _, err := grafeasClient.GrafeasV1Beta1Api.ListNotes(context.Background(), projectID, nil)
	if err != nil {
		log.Fatal(err)
	} else {
		log.Print(notesResp.Notes)
	}
}
```

By adding in the `project.swagger.json` we now _also_ get access to:

```go
package main

import (
	"context"
	"fmt"
	"log"

	grafeasProjectAPI "github.com/psprings/client-go/0.1.4/project"
)

func main() {
	grafeasProjectCfg := &grafeasProjectAPI.Configuration{
		BasePath:      "http://localhost:8080",
		DefaultHeader: make(map[string]string),
		UserAgent:     "Swagger-Codegen/0.1.0/go",
	}
	grafeasProjectClient := grafeasProjectAPI.NewAPIClient(grafeasProjectCfg)
	ctx := context.Background()
	projectID := "projects/myproject"
	createdProject, _, err := grafeasProjectClient.ProjectsApi.CreateProject(ctx, grafeasProjectAPI.ProjectProject{
		Name: projectID,
	})
	if err != nil {
		log.Println(err)
	}
	log.Println(createdProject)
	projectsResp, _, _ := grafeasProjectClient.ProjectsApi.ListProjects(ctx, nil)
	log.Printf("%#v", projectsResp)
}
```

My thought was that it might be a more desirable/uniform experience to merge the spec files in such a way that you only have to do one client import. If I'm off base here, I can remove the `project.swagger.json` import and scrap this logic?

Reply:
Hey @aysylu what do you think about adding the project API to the generated client (to accompany the generated "grafeas" API) per my [comment above](https://github.com/grafeas/client-go/pull/15#discussion_r379539658)?

**Any thoughts on the notion of:**
1. Including the `project.swagger.json` spec in the client generation (at a minimum)

[and/or]

2. _Merging_ the two swagger spec files to form a cohesive client capable with interacting with both APIs?**

Presumably the `Configuration` type will always be the same underlying struct [literal] between the `grafeasAPI` and `grafeasProjectAPI` above.

```go
// Grafeas API Configuration & Client
grafeasCfg := &grafeasAPI.Configuration{
    BasePath:      "http://localhost:8080",
}
grafeasClient := grafeasAPI.NewAPIClient(grafeasCfg)
// Project Configuration & Client
grafeasProjectCfg := &grafeasProjectAPI.Configuration{
    BasePath:      "http://localhost:8080",
}
grafeasProjectClient := grafeasProjectAPI.NewAPIClient(grafeasProjectCfg)
```

Instead of having to do conversion or reflection or just duplicating the `Configuration` to interact with both parts of the API, it seems like it might be more desirable to the average consumer of the client library to interact with a common `grafeasAPI` client?

Reply:
Hi @psprings, thanks for clarifying! I absolutely agree that merging the 2 APIs for the more powerful client is a great idea. And it's a strict improvement over the previous state.